crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -1910,35 +1928,16 @@ where
                         });
                         self.semantic_model.global_scope_mut().add(name, id);
                     }
-                }
-
-                let definition = docstrings::extraction::extract_definition(
-                    ExtractionTarget::Function,
-                    stmt,
-                    self.semantic_model.definition_id,
-                    &self.semantic_model.definitions,
-                );
-                self.semantic_model.push_definition(definition);
-
-                self.semantic_model
-                    .push_scope(ScopeKind::Function(FunctionDef {
-                        name,
-                        body,
-                        args,
-                        decorator_list,
-                        async_: matches!(stmt, Stmt::AsyncFunctionDef(_)),
-                        globals,
-                    }));
 
-                self.deferred.functions.push(self.semantic_model.snapshot());
+                    self.semantic_model.scope_mut().add_global(name, range);
+                }
             }
             Stmt::ClassDef(ast::StmtClassDef {
```

Comment:
Part 1

```suggestion
            Stmt::ClassDef(class_def @ ast::StmtClassDef {
```

Reply:
Ugh, I know, I just hated that it wasn't symmetric with the function case :sob: